src/gf.c

```diff
@@ -2340,11 +2351,12 @@ JL_DLLEXPORT jl_svec_t *jl_match_method(jl_value_t *type, jl_value_t *sig,
 // arguments.
 static int tvar_exists_at_top_level(jl_value_t *tv, jl_tupletype_t *sig, int attop)
```

Comment:
`attop` is hardcoded to one, we could drop that. also could recurse into Tuples and look for TypeVars there in covariant position